src/Transients.cc

```diff
@@ -224,11 +224,12 @@ Transients::addEntry(StoreEntry *e, const cache_key *key, const Store::IoStatus
     slot->set(*e, key);
     e->mem_obj->xitTable.index = index;
     if (direction == Store::ioWriting) {
-        // keep write lock; the caller will decide what to do with it
+        // allow reading and receive remote DELETE events, but do not switch to
+        // the reading lock because transientReaders() callers want true readers
         map->startAppending(e->mem_obj->xitTable.index);
     } else {
         // keep the entry locked (for reading) to receive remote DELETE events
-        map->closeForWriting(e->mem_obj->xitTable.index);
+        map->switchWritingToReading(e->mem_obj->xitTable.index);
```

Comment:
The other use of switchWriting() ToReading explicitly set the xitTable.io to ioReading. I think this one should as well if only to prevent possible changes to the XitTable class and the possible non-ioWriting states from having hidden breakages in future.

Reply:
Done in branch commit e7025c6, along with a couple of other related (and belated) changes.